lib/spack/spack/util/compression.py

```diff
@@ -278,6 +275,208 @@ def decompressor_for(path, ext):
     return _untar
 
 
+class FileType:
```

Comment:
In this case the base class is defining a fair amount of default implementations, so I could raise some `NotImplementedError`s however I would then need to either duplicate a fair amount of code or create a mixin class that essentially is just a copy of the current state of `FileType` which to me propagates the issue. If you had a better way in mind, I am absolutely all ears, because I agree, having `FileType` hanging around is confusing. Docs are updated.